apps/src/pythonlab/pythonHelpers/messageHelpers.ts

```diff
@@ -43,3 +44,32 @@ export function parseErrorMessage(errorMessage: string) {
   const adjustedErrorLines = errorLines.slice(mainErrorLine, errorLines.length);
   return adjustedErrorLines.join('\n');
 }
+
+// This function parses the message string (example: '[PAINTER] PAINT {"color": "Blue"}') to a NeighborhoodSignal.
+export function parseMessageToNeighborhoodSignal(
+  message: string
+): NeighborhoodSignal {
+  /*
+    \[(\w+)]\ captures the signal type inside square brackets, e.g., PAINTER
+    \s+ matches one or more spaces
+    ([^\s]+) captures the value, e.g., PAINT.
+    (\{.*\})? optionally captures the detail json if it exists, e.g., {"color": "Blue"}
+  */
+  const regex = /^\[(\w+)]\s+([^\s]+)(?:\s+(\{.*\}))?$/;
+
+  const match = message.match(regex);
+  if (!match) {
```

Comment:
We should fail quietly rather than throw an error, as we generally want to avoid the page crashing. Can we instead log to cloudwatch with `Lab2MetricsReporter` and return null or something?

Reply:
Sounds good - updated!

Reply:
Added another test for fail to match case.